src/data_model/transformations.jl

```diff
@@ -472,3 +472,917 @@ function apply_voltage_angle_difference_bounds!(eng::Dict{String,<:Any}, vad::Re
+                            v_start[(f_bus,t)] = v_fr[i]
+                            push!(stack, (f_bus,t))
+                        end
+                    end
+                end
+            end
+        end
+        if verbose
+            progress = round(sum(ismissing.(values(v_start)))/length(v_start)*100, digits=2)
+            println("it. $count:\t$progress% left to initialize at end")
+        end
+    end
+
+    # increment non-grounded zero values with epsilon
+    for (k,v) in v_start
+        (b,t) = k
+        data_bus = data_math["bus"][string(b)]
+        grounded_t = Dict(data_bus["terminals"].=>data_bus["grounded"])
+        if !ismissing(v) && iszero(v) && !grounded_t[t]
+            v_start[k] = convert(Complex, epsilon)
+        end
+    end
+
+    return v_start
+end
+
+
+"""
+Adds start values for the voltage to the buses.
+For a multinetwork data model, you can calculate the start voltages for a representative network through 'calc_start_voltage',
+and pass the result as 'uniform_v_start' to use the same values for all networks and avoid recalculating it for each network. 
+The argument 'epsilon' controls the offset added to ungrounded terminals which would otherwise be set to zero.
+"""
+function add_start_voltage!(data_math::Dict{String,Any}; coordinates=:rectangular, uniform_v_start=missing, vr_default=0.0, vi_default=0.0, vm_default=0.0, va_default=0.0, verbose=false, epsilon::Number=1E-3)
+    @assert data_math["data_model"]==MATHEMATICAL
+    @assert coordinates in [:polar, :rectangular] "Legal values for the 'coordinates' argument are [:polar,:rectangular], not :$coordinates."
+    
+    is_mn = haskey(data_math, "multinetwork")
+
+    for (nw,dm) in (is_mn ? data_math["nw"] : [("", data_math)])
+        if ismissing(uniform_v_start)
+            v_start = calc_start_voltage(dm, verbose=verbose, epsilon=epsilon)
+        else
+            v_start = uniform_v_start
+        end
+        for (_, bus) in dm["bus"]
+            index = bus["index"]
+            if coordinates==:rectangular
+                bus["vr_start"] = [ismissing(v_start[(index, t)]) ? vr_default : real(v_start[(index, t)]) for t in bus["terminals"]]
+                bus["vi_start"] = [ismissing(v_start[(index, t)]) ? vi_default : imag(v_start[(index, t)]) for t in bus["terminals"]]
+            elseif coordinates==:polar
+                bus["vm_start"] = [ismissing(v_start[(index, t)]) ? vm_default : abs(v_start[(index, t)]) for t in bus["terminals"]]
+                bus["va_start"] = [ismissing(v_start[(index, t)]) ? va_default : angle(v_start[(index, t)]) for t in bus["terminals"]]
+            end
+        end
+    end
+
+    return data_math
+end
+
+
+"Short-hand for add_start_voltage! with rectangular coordinates."
+add_start_vrvi!(data_math::Dict{String,Any}; kwargs...) = add_start_voltage!(data_math, coordinates=:rectangular, kwargs...)
+
+
+"Short-hand for add_start_voltage! with polar coordinates."
```

Comment:
For exported functions, we should follow the new docstring pattern where the function is given (indented) first, see, e.g. transform_data_model function as an example.